File: challenges/code_eval/medium/endianness/endianness.go
package main

import (
	"fmt"
	"unsafe"
)

// Endianness
// https://www.codeeval.com/open_challenges/15/
func main() {
	var i int32 = 0x01020304
	u := (*byte)(unsafe.Pointer(&i))

	if *u == 0x04 {
		fmt.Println("LittleEndian")
	} else {
		fmt.Println("BigEndian")
	}
}

File: challenges/code_eval/medium/endianness.c
#include "stdio.h"

// Endianness
// https://www.codeeval.com/open_challenges/15/
int main() {
    int n = 1;

    if (*(char *)&n == 1) {
        printf("LittleEndian\n");
    } else {
        printf("BigEndian\n");
    }
}
